cdc/processor/pipeline/sorter.go

```diff
@@ -120,20 +123,85 @@ func (n *sorterNode) Init(ctx pipeline.NodeContext) error {
 		metricsTicker := time.NewTicker(flushMemoryMetricsDuration)
 		defer metricsTicker.Stop()
 
+		var (
+			eventToMount *model.PolymorphicEvent
+			nextEvent    *model.PolymorphicEvent
+			dummyDoneCh  chan struct{}
+		)
+		dummyDoneCh = make(chan struct{})
+		close(dummyDoneCh)
+
+		nextReady := func() <-chan struct{} {
+			if nextEvent == nil {
+				return nil
+			}
+			if nextEvent.RawKV.OpType != model.OpTypeResolved {
+				ret := nextEvent.Finished()
+				if ret == nil {
+					log.Panic("empty finished ch")
+				}
+				return ret
+			}
+			return dummyDoneCh
+		}
+
+		loadNext := func() <-chan *model.PolymorphicEvent {
+			if nextEvent != nil {
+				return nil
+			}
+			return n.mounterWaitCh
+		}
+
+		writeNext := func() chan<- *model.PolymorphicEvent {
+			if eventToMount == nil {
+				return nil
+			}
+			return n.mounterWaitCh
+		}
+
+		readFromSorter := func() <-chan *model.PolymorphicEvent {
+			if eventToMount != nil {
+				return nil
+			}
+			return sorter.Output()
+		}
+
 		for {
 			select {
 			case <-stdCtx.Done():
 				return nil
 			case <-metricsTicker.C:
 				metricsTableMemoryGauge.Set(float64(n.flowController.GetConsumption()))
-			case msg, ok := <-sorter.Output():
+			case msg, ok := <-readFromSorter():
 				if !ok {
 					// sorter output channel closed
 					return nil
 				}
 				if msg == nil || msg.RawKV == nil {
 					log.Panic("unexpected empty msg", zap.Reflect("msg", msg))
 				}
+
+				// resolved events do not need to be mounted
+				if msg.RawKV.OpType != model.OpTypeResolved {
+					msg.SetUpFinishedChan()
+					select {
+					case <-ctx.Done():
+						return nil
+					case n.mounter.Input() <- msg:
+					}
+				}
+				eventToMount = msg
+			case writeNext() <- eventToMount:
+				eventToMount = nil
+			case msg, ok := <-loadNext():
```

Comment:
Could you select the two cases in another `select` statement (with `default` if no-blocking is needed)?